### PR TITLE
[FLINK-29907] Externalize Kinesis/Firehose documentation

### PR DESCRIPTION
**Pull request by dannycranmer**
Copy over documentation for Kinesis and Firehose from the Flink repo

### CODE REVIEW

File: docs/content.zh/docs/connectors/datastream/firehose.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        .setMaxTimeInBufferMS(5000)                       // Optional
+        .setMaxRecordSizeInBytes(1000 * 1024)             // Optional
+        .build();
+
+flinkStream.sinkTo(kdfSink);
+```
+{{< /tab >}}
+{{< tab "Scala" >}}
+```scala
+Properties sinkProperties = new Properties()
```
Comment (hlteoh37) [inline docs/content.zh/docs/connectors/datastream/firehose.md:71]: nit: `val`

File: docs/content.zh/docs/connectors/table/kinesis.md [hunk 2]
```diff
@@ -0,0 +1,914 @@
...
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Amazon Kinesis Data Streams SQL Connector
+
+{{< label "Scan Source: Unbounded" >}}
+{{< label "Sink: Streaming Append Mode" >}}
+
```
Comment (hlteoh37) [inline docs/content.zh/docs/connectors/table/kinesis.md:31]: nit: add batch?

File: docs/content.zh/docs/connectors/datastream/firehose.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+The Firehose sink writes to [Amazon Kinesis Data Firehose](https://aws.amazon.com/kinesis/data-firehose/).
+
+Follow the instructions from the [Amazon Kinesis Data Firehose Developer Guide](https://docs.aws.amazon.com/firehose/latest/dev/basic-create.html)
+to setup a Kinesis Data Firehose delivery stream.
+
+To use the connector, add the following Maven dependency to your project:
+
+{{< artifact flink-connector-aws-kinesis-firehose >}}
+
+{{< connector_artifact flink-connector-aws-kinesis-firehose 4.0.0 >}}
```
Comment (hlteoh37) [inline docs/content.zh/docs/connectors/datastream/firehose.md:38]: should we add 
```
{{< py_download_link "aws-kinesis-firehose" >}}
```
like the non-chinese docs?

File: docs/content/docs/connectors/datastream/firehose.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
+---
+title: Firehose
+weight: 5
+type: docs
+aliases:
+- /dev/connectors/firehose.html
```
Comment (hlteoh37) [inline docs/content/docs/connectors/datastream/firehose.md:6]: not sure how this works, but should we add the below like kinesis data streams docs?
```
  - /apis/streaming/connectors/firehose.html
```

File: docs/content.zh/docs/connectors/table/firehose.md [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Amazon Kinesis Data Firehose SQL Connector
+
+{{< label "Sink: Streaming Append Mode" >}}
```
Comment (hlteoh37) [inline docs/content.zh/docs/connectors/table/firehose.md:30]: nit: add batch?

File: docs/content/docs/connectors/datastream/firehose.md (hunk 4 above)
Comment (dannycranmer) [reply docs/content/docs/connectors/datastream/firehose.md:6]: I think this might be a legacy thing, not too sure tbh

**Pull request merged**